src/mol-repr/structure/complex-visual.ts

```diff
@@ -198,17 +200,30 @@ export function ComplexVisual<G extends Geometry, P extends StructureParams & Ge
         }
     }
 
+    function finalize(ctx: VisualContext) {
+        if (renderObject) {
+            processValues?.(renderObject.values, geometry, currentProps, currentTheme, ctx.webgl);
+        }
+    }
+
     return {
         get groupCount() { return locationIt ? locationIt.count : 0; },
         get renderObject () { return locationIt && locationIt.count ? renderObject : undefined; },
         createOrUpdate(ctx: VisualContext, theme: Theme, props: Partial<PD.Values<P>> = {}, structure?: Structure) {
             prepareUpdate(theme, props, structure || currentStructure);
             if (updateState.createGeometry) {
                 const newGeometry = createGeometry(ctx, newStructure, newTheme, newProps, geometry);
-                return newGeometry instanceof Promise ? newGeometry.then(update) : update(newGeometry);
+                if (newGeometry instanceof Promise) {
```

Comment:
I know this was here before, but in general it might be safer to use a helper function akin to

```ts
function isPromiseLike<T = any>(x: any): x is Promise<T> {
  return typeof x?.then === 'function';
}
```

in the case ``Promise`` comes from an unknown library. There seem to be 3 instance of this ``instanceof Promise`` in ``mol-repr``.

Reply:
Good point, added the helper